torch/csrc/jit/autodiff.cpp

```diff
@@ -25,22 +24,45 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
   return fmap(sym_grads, [](const SymbolicVariable &v) { return v.value(); });
 }
 
-void differentiate(std::shared_ptr<Graph>& graph) {
+void differentiate(std::shared_ptr<Graph>& graph,
+                   ArrayRef<bool> requires_grad_inputs,
+                   ArrayRef<bool> requires_grad_outputs) {
```

Comment:
We can only have an array for inputs. We compute whether we an output requires a gradient just like the how the autograd already does it so that we precisely match its behavior.